packages/shopping-cart/src/App.tsx

```diff
@@ -31,13 +43,15 @@ function App() {
           </div>
         </div>
       </div>}
-      <LoadingContext.Provider value={{ isLoading, updateLoading }}>
-        <UsersContext.Provider value={{users, addUser}}>
-          <AuthContext.Provider value={{authStatus, updateStatus}}>
-              <Header/>
-          </AuthContext.Provider>
-        </UsersContext.Provider>
-      </LoadingContext.Provider>
+      <CartContext.Provider value={{ cart, addToCart, updateCart }}>
```

Comment:
again bad idea, provide all context at app level what's the point in separating them? why does the register page need `carContext` and so on?